builtin/builtin.mbti

```diff
@@ -190,8 +190,11 @@ impl Iter {
   find_first[T](Self[T], (T) -> Bool) -> T?
   flat_map[T, R](Self[T], (T) -> Self[R]) -> Self[R]
   fold[T, B](Self[T], ~init : B, (B, T) -> B) -> B
+  head[A](Self[A]) -> A?
+  intersperse[A](Self[A], A) -> Self[A]
```

Comment:
This name is kinda confusing (at least for me though).

Maybe `insert_seperator` or `interleave_with`?

Reply:
I use this name to align with `interperse`/`intercalate` in `@immut/list` #63 . [Rust](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.intersperse) also use `intersperse` in iterator.